### ISSUE TITLE
build failure on OS X 10.12.1 Beta (16B2553a)

### ISSUE THREAD

**Issue by lcowell** [redox-os/redox#719]
**Reproduction**: [describe how you are able to reproduce ("trigger") this bug/issue.]
```
rustup update
(master) rustc -V
rustc 1.14.0-nightly (f09420685 2016-10-20)

make clean
git clean -X -f -d
git remote add upstream git@github.com:redox-os/redox.git
git pull upstream master
git submodule update --recursive --init
make all

RUST_BACKTRACE=1 rustc --target=i386-unknown-redox.json --cfg redox -L build/i386-unknown-redox/debug -C no-prepopulate-passes -C no-stack-check -C opt-level=3 -Z no-landing-pads -A dead_code -C ar=i386-elf-ar -C linker=i386-elf-gcc -o build/i386-unknown-redox/debug/libstd.rlib libstd/src/lib.rs -L native=libc/lib/
error: static variable `__errno` should have an upper case name such as `__ERRNO`
 --> libstd/src/externs.rs:3:1
  |
3 | pub static mut __errno: isize = 0;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: lint level defined here
 --> libstd/src/lib.rs:46:9
  |
46| #![deny(warnings)]
  |         ^^^^^^^^

error: static variable `_args` should have an upper case name such as `_ARGS`
  --> libstd/src/env.rs:20:1
   |
20 | static mut _args: *mut Vec<&'static str> = 0 as *mut Vec<&'static str>;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> libstd/src/lib.rs:46:9
   |
46 | #![deny(warnings)]
   |         ^^^^^^^^

error: aborting due to 2 previous errors

make: *** [build/i386-unknown-redox/debug/libstd.rlib] Error 101
```
**Expected behavior**: [describe the behavior you would expect the repro to yield.]
it compiles

**Actual behavior**: [describe the actual behavior, which is presented through the repro.].
It exits early

```
echo $?
2
```

**Build information**: [output of `rustc -V`, `git rev-parse HEAD`, `qemu-i386 -version`, `uname -a`, etc.]

```
rustc -V
rustc 1.14.0-nightly (f09420685 2016-10-20)

git rev-parse HEAD
1bd3737fc9c951a028efa1c9686654f0f1d973ea

uname -a
Darwin asdf.local 16.1.0 Darwin Kernel Version 16.1.0: Mon Oct  3 14:03:15 PDT 2016; root:xnu-3789.21.3~20/RELEASE_X86_64 x86_64

OS X version
10.12.1 Beta (16B2553a)
```

**Blocking/related**: [issues or PRs blocking or being related to this issue.]
I didn't find any issues listed with the error message in the build output.
There are no open PRs

**Misc**: [optional: for other relevant information that should be known or cannot be described in the other fields.]

------

_If the above does not fit the nature of the issue feel free to modify it._

**Comment by cyplo** [comment]
Same errors here on Fedora 24 as well as on Travis (https://travis-ci.org/redox-os/redox/builds/169549287) Maybe different linter settings for the main repo and libstd ?

**Comment by Yoric** [comment]
This has been fixed.

**Comment by cyplo** [comment]
Fixed for me as of e40b4720b671e177807d394842eaac116551342b, thank you !
